### PR TITLE
Check the region only when creating web app

### PR DESCRIPTION
**Pull request by SummerSun**
Remove the `createOrUpdateWebApp`, separate the create Web App logic and update Web App logic and move them into the `doExecute()` for better readability.

Only when creating a new Web App , the region is required.

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/V2ConfigurationParser.java [hunk 1]
```diff
@@ -51,9 +51,6 @@ protected OperatingSystemEnum getOs() throws MojoExecutionException {
     @Override
     protected Region getRegion() throws MojoExecutionException {
         final String region = mojo.getRegion();
-        if (StringUtils.isEmpty(region)) {
-            throw new MojoExecutionException("Please config the <region> in pom.xml.");
-        }
         if (Arrays.asList(Region.values()).contains(region)) {
```
Comment (Flanker32) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/V2ConfigurationParser.java]: !contains()

**Comment by SummerSun** [comment]
test

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/V2ConfigurationParser.java [hunk 2]
```diff
@@ -51,7 +50,7 @@ protected OperatingSystemEnum getOs() throws MojoExecutionException {
     @Override
     protected Region getRegion() throws MojoExecutionException {
         final String region = mojo.getRegion();
-        if (!Arrays.asList(Region.values()).contains(region)) {
+        if (Region.findByLabelOrName(region) == null) {
```
Comment (Flanker32) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/V2ConfigurationParser.java]: What will happen if region is not config in pom here? will it also throw region not support Exception?

Comment (SummerSun) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/V2ConfigurationParser.java]: My fault. That case should be taken care of.

**Pull request merged**